pkg/cmd/kafka/acl/consumer/consumer.go

```diff
@@ -0,0 +1,216 @@
+package consumer
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/redhat-developer/app-services-cli/internal/config"
+	"github.com/redhat-developer/app-services-cli/pkg/cmd/factory"
+	"github.com/redhat-developer/app-services-cli/pkg/connection"
+	"github.com/redhat-developer/app-services-cli/pkg/iostreams"
+	"github.com/redhat-developer/app-services-cli/pkg/localize"
+	"github.com/redhat-developer/app-services-cli/pkg/logging"
+	"github.com/spf13/cobra"
+
+	kafkainstanceclient "github.com/redhat-developer/app-services-sdk-go/kafkainstance/apiv1internal/client"
+)
+
+type options struct {
+	Config     config.IConfig
+	Connection factory.ConnectionFunc
+	Logger     logging.Logger
+	IO         *iostreams.IOStreams
+	localizer  localize.Localizer
+	Context    context.Context
+
+	kafkaID    string
+	topic      string
+	user       string
+	svcAccount string
+}
+
+// NewConsumerACLCommand creates ACL rules to allow principal to consume from topics
+func NewConsumerACLCommand(f *factory.Factory) *cobra.Command {
+
+	opts := &options{
+		Config:     f.Config,
+		Connection: f.Connection,
+		Logger:     f.Logger,
+		IO:         f.IOStreams,
+		localizer:  f.Localizer,
+		Context:    f.Context,
+	}
+
+	cmd := &cobra.Command{
+		Use:     "consumer",
+		Short:   f.Localizer.MustLocalize("kafka.acl.consumer.cmd.shortDescription"),
+		Long:    f.Localizer.MustLocalize("kafka.acl.consumer.cmd.longDescription"),
+		Example: f.Localizer.MustLocalize("kafka.acl.consumer.cmd.example"),
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasKafka() {
+				return opts.localizer.MustLocalizeError("kafka.acl.common.error.noKafkaSelected")
+			}
+
+			opts.kafkaID = cfg.Services.Kafka.ClusterID
+
+			if opts.user == "" && opts.svcAccount == "" {
+				return opts.localizer.MustLocalizeError("kafka.acl.common.error.noPricipalsSelected")
+			}
+
+			if opts.user != "" && opts.svcAccount != "" {
+				return opts.localizer.MustLocalizeError("kafka.acl.common.error.bothPricipalsSelected")
+			}
+
+			return runConsumer(opts)
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.user, "user", "", opts.localizer.MustLocalize("kafka.acl.common.flag.user.description"))
+	cmd.Flags().StringVar(&opts.user, "service-account", "", opts.localizer.MustLocalize("kafka.acl.common.flag.serviceAccount.description"))
```

Comment:
I know this is what is described in the doc, but I do fear that is brings complexity by having too many options.

User:

```shell
rhoas kafka acl consumer --principal User::ephelan_kafka_devexp
```

ServiceAccount:

```shell
rhoas kafka acl consumer --principal ServiceAccount::ephelan_kafka_devexp
```

Maybe we can discuss this on the call today.